homeassistant/components/rest_command/__init__.py

```diff
@@ -58,6 +60,19 @@
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the REST command component."""
 
+    async def reload_service_handler(service: ServiceCall) -> None:
+        """Remove all rest_commands and load new ones from config."""
+        conf = await async_integration_yaml_config(hass, DOMAIN) or {}
```

Comment:
This pattern is used in mqtt, but I'm not sure it's wanted. `async_integration_yaml_config` returns `None` if the configuration can't be parsed, for example if there's a yaml synatx error.

I think it's better to return early if `async_integration_yaml_config` and if not clean everything up.